Library/Homebrew/tab.rb

```diff
@@ -99,42 +69,127 @@ def self.from_file_content(content, path)
       raise e, "Cannot parse #{path}: #{e}", e.backtrace
     end
     attributes["tabfile"] = path
-    attributes["source_modified_time"] ||= 0
-    attributes["source"] ||= {}
 
-    tapped_from = attributes["tapped_from"]
-    if !tapped_from.nil? && tapped_from != "path or URL"
-      attributes["source"]["tap"] = attributes.delete("tapped_from")
-    end
+    new(attributes)
+  end
 
-    if attributes["source"]["tap"] == "mxcl/master" ||
-       attributes["source"]["tap"] == "Homebrew/homebrew"
-      attributes["source"]["tap"] = "homebrew/core"
-    end
+  def self.empty
+    attributes = {
+      "homebrew_version"        => HOMEBREW_VERSION,
+      "installed_as_dependency" => false,
+      "installed_on_request"    => false,
+      "loaded_from_api"         => false,
+      "time"                    => nil,
+      "runtime_dependencies"    => nil,
+      "arch"                    => nil,
+      "source"                  => {
+        "path"         => nil,
+        "tap"          => nil,
+        "tap_git_head" => nil,
+      },
+      "built_on"                => DevelopmentTools.generic_build_system_info,
+    }
+
+    new(attributes)
+  end
+
+  def self.tap_git_head(formula_or_cask)
+    return unless formula_or_cask.tap&.installed?
+
+    formula_or_cask.tap.git_head
+  end
```

Comment:
Idea: Maybe we could use the `Formula#tap_git_head` and `Cask::Cask#tap_git_head` methods instead so that we include this information from the API as well. It might not even be necessary to have a separate method here anymore in that case.

Reply:
Good catch, didn't realize those existed!